appspec-scripts/start_app.sh

```diff
@@ -1,10 +1,25 @@
 #!/bin/bash
-set -e
-export PATH=$PATH:/home/ubuntu/.cargo/bin
+set -x
 
+TAG=$(curl http://169.254.169.254/latest/meta-data/tags/instance/stable)
 APP_NAME=$(echo $APPLICATION_NAME | awk -F- '{ print $2 }')
 
-cd ~/$APP_NAME/tlsn/notary-server
-target/release/notary-server --config-file ~/.notary/$APP_NAME/config.yaml &> ~/$APP_NAME/tlsn/notary.log &
+if [ $APP_NAME = "stable" ]; then
+  STABLE_PORTS="7047 7057 7067"
+  for PORT in $STABLE_PORTS; do
+    PORT_LISTENING=$(netstat -lnt4 | egrep -cw $PORT)
+    if [ $PORT_LISTENING -eq 0 ]; then
+      cd ~/${APP_NAME}_${TAG}/tlsn/notary-server
+      target/release/notary-server --config-file ~/.notary/${APP_NAME}_${PORT}/config.yaml &> ~/${APP_NAME}_${TAG}/tlsn/notary.log &
+      # Create a tag that will be used for service validation
+      INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+      aws ec2 create-tags --resources $INSTANCE_ID --tags "Key=port,Value=$PORT"
+    break
+    fi
```

Comment:
if no port is available, shouldn't we also exit the script and fail the deployment since something must have gone wrong?

Reply:
actually wouldn't the validation still pass? assuming that the instance port tag is always assigned with a value (since 3 versions are running since the beginning), and the oldest stable service fails to be killed (similar to  https://github.com/tlsnotary/tlsn/pull/459#discussion_r1533286846), then the validation check will still able to detect that a stable service is still listening at the port?

Reply:
maybe one way to solve this is in validation check, we ensure that version of the stable service that is running at the port matches the instance version tag (since that one is guaranteed to be true)

Reply:
and for the nightly problem (https://github.com/tlsnotary/tlsn/pull/459#discussion_r1533286846), since we don't have version there, maybe validation needs to ensure that the nightly running at that port is not older than X hours/day?

Reply:
Please check: https://github.com/tlsnotary/tlsn/pull/459#discussion_r1533325517